core/model/src/main/java/org/eclipse/rdf4j/model/util/LexicalValueComparator.java

```diff
@@ -94,7 +94,7 @@ private int compareBNodes(BNode leftBNode, BNode rightBNode) {
 	}
 
 	private int compareURIs(IRI leftURI, IRI rightURI) {
-		return leftURI.toString().compareTo(rightURI.toString());
+		return leftURI.compareTo(rightURI);
```

Comment:
The contract of the LexicalValueComparator is that it compares on the lexical value of the RDF term. This change is not valid because you don't know if the IRI implementations actually use lexical comparison internally. 

Except of course if, as mentioned above, we explicitly document that IRI should always implement Comparable using lexical value comparison.

Reply:
Both out comparators use the same way of comparing IRIs.